File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives (self, grid : List [List [int]]) -> int:
        count = 0
        i, j = 0, len (grid [0]) - 1
        while i < len (grid) and j >= 0:
            if grid [i] [j] < 0:
                count += (len (grid) - i)
                j -= 1
            else: i += 1
        return count

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.java
class Solution {
    public int countNegatives (int [] [] grid) {
        int count = 0;
        for (int i = 0, j = grid [0].length - 1; i < grid.length && j >= 0;) {
            if (grid [i] [j] < 0) {
                count += (grid.length - i);
                j--;
            }
            else i++;
        }
        return count;
    }
}